src/libAtomVM/context.c

```diff
@@ -202,14 +194,23 @@ size_t context_size(Context *ctx)
         + context_memory_size(ctx) * BYTES_PER_TERM;
 }
 
+static inline size_t context_heap_size(const Context *ctx)
+{
+    size_t heap_size = ctx->heap.heap_ptr - ctx->heap.heap_start;
+    if (ctx->heap.next) {
+        heap_size += memory_heap_fragment_memory_size(ctx->heap.next);
+    }
+    return heap_size;
+}
+
 bool context_get_process_info(Context *ctx, term *out, term atom_key)
 {
-    if (UNLIKELY(memory_ensure_free(ctx, 3) != MEMORY_GC_OK)) {
+    if (UNLIKELY(memory_ensure_free_maybe_gc(ctx, 3) != MEMORY_GC_OK)) {
```

Comment:
I have some general comments about these _maybe_gc name changes.  I am personally against them, unless the signature is changing as well.  I have come to understand when these functions can result in a gc, so I don't find them necessary.  For new developers it might help, but there is so much else to get tripped up on I don't know if it is worth making this change, because it has a big impact on downstream drivers.